src/rust/ide/lib/analytics/src/remote_log.rs

```diff
@@ -0,0 +1,21 @@
+//! Provides an API to send data to our remote logging service. Requires the remote logging
+//! to be already set up on the JS side. That means, there needs to exist a `window.enso.remote_log`
+//! method that takes a string and does the actual logging.
+
+use js_sys;
+use wasm_bindgen::JsValue;
+
+/// Send the provided public event to our logging service.
+pub fn remote_log(message:crate::data::Public) {
+    let js_new  = js_sys::Function::new_with_args("msg", "window.enso.remote_log(msg)");
+    let context = JsValue::NULL;
+    let _res = js_new.call1(&context,&message.data.into());
+}
+
+/// Send the provided public event with data to our logging service.
+pub fn remote_log_data(message:crate::data::Public,data:crate::data::Public) {
```

Comment:
I would add here checking (if logging string-like things) if the data is bigger than lets say 50 chars. If so, lets cut it down and print warning. The sae applies to event name

Reply:
I added this functionality on the JS side, so we can keep data non-stringified as long as possible.